### PR TITLE
[RISCV] Add mvendorid/marchid/mimpid to CPU definitions

### PR DESCRIPTION
**Pull request by wangpc-pp**
We can get these information via `sys_riscv_hwprobe`.

This can be used to implement `__builtin_cpu_is`.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVProcessors.td [hunk 1]
```diff
@@ -435,7 +438,11 @@ def VENTANA_VEYRON_V1 : RISCVProcessorModel<"veyron-v1",
                                               TuneZExtHFusion,
                                               TuneZExtWFusion,
                                               TuneShiftedZExtWFusion,
-                                              TuneLDADDFusion]>;
+                                              TuneLDADDFusion]> {
+  let MVendorID = 0x61f;
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVProcessors.td:442]: This is from https://github.com/qemu/qemu/blob/master/target/riscv/cpu_vendorid.h.

**Review by lenary — APPROVED** [review]

File: llvm/lib/Target/RISCV/RISCVProcessors.td [hunk 2]
```diff
@@ -435,7 +438,11 @@ def VENTANA_VEYRON_V1 : RISCVProcessorModel<"veyron-v1",
...
+  let MArchID = 0x8000000000010000;
+  let MImpID = 0x111;
```
Comment (dtcxzyw) [inline llvm/lib/Target/RISCV/RISCVProcessors.td:444]: For spacemit-k1:
```
mvendorid       : 0x710
marchid         : 0x8000000058000001
mimpid          : 0x1000000049772200
```

**Review by dtcxzyw — APPROVED** [review]
LGTM.

File: llvm/include/llvm/TargetParser/RISCVTargetParser.h [hunk 3]
```diff
@@ -32,6 +32,21 @@ struct RISCVExtensionBitmask {
 };
 } // namespace RISCVExtensionBitmaskTable
 
+struct CPUModel {
+  uint32_t MVendorID;
+  uint64_t MArchID;
+  uint64_t MImpID;
```
Comment (topperc) [inline llvm/include/llvm/TargetParser/RISCVTargetParser.h:38]: MImplID to match the name of the CSR?

Comment (topperc) [reply llvm/include/llvm/TargetParser/RISCVTargetParser.h:38]: Nevermind, the CSR is mimpid. hwprobe is the one that added the 'l' to the name.

**Review by topperc — APPROVED** [review]
LGTM

**Comment by wangpc-pp** [comment]
b36fcf4f493ad9d30455e178076d91be99f3a7d8 should fix the error, sorry for the noise.

**Pull request merged**